src/steamship/langchain/llm/llm.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import tempfile
+from typing import Any, List, Mapping, Optional
+
+from langchain.llms.base import LLM
+from transformers import GPT2TokenizerFast
+
+from steamship import Steamship
+from steamship.data import TagKind, TagValueKey, tag_utils
+
+
+class SteamshipLLM(LLM):
+    """Implements LangChain LLM interface in a Steamship-compatible fashion, allowing use in chains/agents as required.
+
+    NOTE: This provides a **synchronous** interaction with the LLM backend.
+    """
+
+    client: Steamship
+    plugin_handle: str = "gpt-3"
+    temperature: float = 0.8
+    max_words: int = 500
+
+    @property
+    def _llm_type(self) -> str:
+        return f"steamship-{self.plugin_handle}"
+
+    def _call(self, prompt: str, stop: Optional[List[str]] = None) -> str:
+        stop_str = ",".join(stop) if stop is not None else ""
+        llm_config = {
+            "temperature": self.temperature,
+            "max_words": self.max_words,
+            "stop": stop_str,
+        }
+        instance_handle = f"gpt-3-{''.join(filter(str.isalnum, stop_str)).lower()}"
+
+        # we create a plugin instance in `_call` because currently `stop` params are passed at configuration-time,
+        # not run-time
+        llm_plugin = self.client.use_plugin(
+            plugin_handle=self.plugin_handle,
+            instance_handle=instance_handle,
+            config=llm_config,
+            fetch_if_exists=True,
+        )
+
+        logging.info(f"invoking {self.plugin_handle} with prompt: {prompt}")
+        task = llm_plugin.tag(doc=prompt)
```

Comment:
Might be helpful to use the `llm_plugin.generate` call here to synchronously go from prompt to output.

```
plugin = self.client.use_plugin(...)
llm_plugin = cast(PromptGenerationPluginInstance, plugin)
generation = llm_plugin.generate(prompt_text, prompt_variables)
```

Reply:
Good point. I copy/pasted this from earlier stabs in the dark that predated that. I'll update.

Reply:
Looking at this again. This isn't a strictly perfect update as the `prompt` string from langchain is the populated template (vs the template itself). Put another way, we don't have the `prompt_variables` at this point. But maybe that's OK and we just ignore that for now?